src/skirmishbunker/Bunker.py

```diff
@@ -42,60 +44,28 @@ def __init__(self):
         init_floor_cut(self)
         init_pip_params(self)
 
-    def make_series(self, shape, length_offset, x_translate=0, y_translate=0, z_translate=0, skip_list=None, keep_list=None):
-        length = self.int_length
-        width = self.int_width
-        padding = self.panel_padding
-        p_width = self.panel_width
-
-        x_panels_size = math_floor(length / (self.panel_length + self.panel_padding))
-        y_panels_size = math_floor(width / (self.panel_length + self.panel_padding))
-
-        x_shapes = series(shape, x_panels_size, length_offset=length_offset)
-        y_shapes = series(shape, y_panels_size, length_offset=length_offset)
-
-        x_plus = (
-            cq.Workplane("XY").add(x_shapes)
-            .translate((0, y_translate, z_translate))
-        )
-
-        x_minus = (
-            cq.Workplane("XY").add(x_shapes)
-            .rotate((0,0,1),(0,0,0),180)
-            .translate((0, -1*y_translate, z_translate))
-        )
-
-        y_plus = (
-            cq.Workplane("XY").add(y_shapes)
-            .rotate((0,0,1),(0,0,0),90)
-            .translate((x_translate, 0, z_translate))
-        )
-
-        y_minus = (
-            cq.Workplane("XY").add(y_shapes)
-            .rotate((0,0,1),(0,0,0),90)
-            .rotate((0,0,1),(0,0,0),180)
-            .translate((-1*(x_translate), 0, z_translate))
-        )
-
-        scene = x_plus.add(y_plus).add(x_minus).add(y_minus)
-
-        if skip_list and len(skip_list) > 0:
-            solids = scene.solids().vals()
-            scene = cq.Workplane("XY")
-
-            for  index, solid in enumerate(solids):
-                if index not in skip_list:
-                    scene.add(solid)
-        elif keep_list and len(keep_list) > 0:
-            solids = scene.solids().vals()
-            scene = cq.Workplane("XY")
-
-            for  index, solid in enumerate(solids):
-                if index in keep_list:
-                    scene.add(solid)
-
-        return scene
+    def make_series(self, shape, length_offset, x_translate = 0, y_translate = 0, z_translate = 0, skip_list = None, keep_list = None):
+        series = SeriesHelper()
+        series.shape = shape
+        series.outer_length = self.int_length
+        series.outer_width = self.int_width
+        series.length_offset = length_offset
+        series.comp_length = self.panel_length
+        series.comp_padding = self.panel_padding
+        series.x_translate = x_translate
+        series.y_translate = y_translate
+        series.z_translate = z_translate
+        series.skip_list = skip_list
+        series.keep_list = keep_list
+        series.make()
+
+        return series.get_scene()
+
+    def set_roof_object(self, roof):
+        if not isinstance(roof, FlatRoof):
```

Comment:
Dig that idea, makes sure the object has an expected baseline of implementation. 
Effectively an interface

Reply:
Oh yea, I totally forgot about interfaces... 🤯